src/qengine/opencl.cpp

```diff
@@ -1898,35 +1896,43 @@ bool QEngineOCL::ApproxCompare(QEngineOCLPtr toCompare)
         otherStateVec = toCompare->stateVec;
         otherStateBuffer = toCompare->stateBuffer;
     } else {
-        otherStateVec = toCompare->AllocStateVec(toCompare->maxQPower, true);
+        otherStateVec = AllocStateVec(toCompare->maxQPower, true);
         toCompare->LockSync(CL_MAP_READ);
         std::copy(toCompare->stateVec, toCompare->stateVec + toCompare->maxQPower, otherStateVec);
         toCompare->UnlockSync();
-        otherStateBuffer = toCompare->MakeStateVecBuffer(otherStateVec);
+        otherStateBuffer = std::make_shared<cl::Buffer>(context, CL_MEM_USE_HOST_PTR | CL_MEM_READ_WRITE, sizeof(complex) * toCompare->maxQPower, otherStateVec);
     }
 
-    QueueCall(OCL_API_APPROXCOMPARE, nrmGroupCount, nrmGroupSize, { stateBuffer, otherStateBuffer, ulongBuffer, nrmBuffer }, sizeof(real1) * nrmGroupSize);
-
-    EventVecPtr waitVec2 = ResetWaitEvents();
+    QueueCall(OCL_API_APPROXCOMPARE, nrmGroupCount, nrmGroupSize,
+        { stateBuffer, otherStateBuffer, ulongBuffer, nrmBuffer }, sizeof(real1) * nrmGroupSize);
 
     real1 sumSqrErr;
-    WAIT_REAL1_SUM(waitVec2, *nrmBuffer, nrmGroupCount / nrmGroupSize, nrmArray, &sumSqrErr);
+    WAIT_REAL1_SUM(*nrmBuffer, nrmGroupCount / nrmGroupSize, nrmArray, &sumSqrErr);
 
     if (toCompare->deviceID != deviceID) {
-        free(otherStateVec);
+        FreeAligned(otherStateVec);
     }
 
     return sumSqrErr < approxcompare_error;
 }
 
 QInterfacePtr QEngineOCL::Clone()
 {
+    clFinish();
+
     QEngineOCLPtr copyPtr = std::make_shared<QEngineOCL>(
         qubitCount, 0, rand_generator, complex(ONE_R1, ZERO_R1), doNormalize, randGlobalPhase, useHostRam, deviceID);
 
-    clFinish();
+	copyPtr->clFinish();
```

Comment:
Whitespace change?

Reply:
Looks like a bunch of these got transitioned from 4 space indent to one tab indent.  Probably need to re-run the style tool.

Reply:
Yes, Visual Studio kept telling me that it was applying our `clang-format` definition, but it apparently screwed it up. I ran `make format` on Linux, just now.